src/cli.js

```diff
@@ -1,37 +1,11 @@
 #!/usr/bin/env node
 import yargs from 'yargs'
-import fs from 'fs'
 import { hideBin } from 'yargs/helpers'
-import { upload } from './upload.js'
+import upload from './commands/upload.js'
 
 yargs(hideBin(process.argv))
   .demandCommand(1, 'You need to specify a command')
-  .command(
-    'upload <directory>',
-    'Upload a directory to Arweave as bundle',
-    (yargs) => {
-      yargs.positional('directory', {
-        describe: 'The directory containing files to upload',
-        type: 'string',
-      })
-    },
-    async (argv) => {
-      const { directory, dryRun } = argv
-      // console.log({ directory, privateKey, dryRun })
-      let privateKey
-      try {
-        privateKey = argv.privateKey
-          ? JSON.parse(Buffer.from(argv.privateKey, 'base64').toString('utf-8'))
-          : JSON.parse(fs.readFileSync('wallet.json'))
-      } catch (e) {
-        throw new Error(
-          'missing private key as base64 of a wallet.json or missing wallet.json',
-        )
-      }
-
-      return await upload(directory, privateKey, dryRun)
-    },
-  )
+  .command(upload.command, upload.description, upload.builder, upload.handler)
   .option('private-key', {
```

Comment:
Please try to move both the options in the `builder` function in the upload command.